docs/content/integrations/duckdb/reference.mdx

```diff
@@ -0,0 +1,584 @@
+
+The SnowflDuckDBake I/O manager can also store data partitioned on multiple dimensions. To do this, you must specify the column for each partition as a dictionary of `partition_expr` metadata:
+
+```python file=/integrations/duckdb/multi_partition.py startafter=start_example endbefore=end_example
+import pandas as pd
+
+from dagster import (
+    DailyPartitionsDefinition,
+    MultiPartitionsDefinition,
+    StaticPartitionDefinition,
+    asset,
+)
+
+
+@asset(
+    partitions_def=MultiPartitionsDefinition(
+        {
+            "date": DailyPartitionsDefinition(start_date="2023-01-01"),
+            "species": StaticPartitionDefinition(
+                ["Iris-setosa", "Iris-virginica", "Iris-versicolor"]
+            ),
+        }
+    ),
+    metadata={"partition_expr": {"date": "TO_TIMESTAMP(TIME)", "species": "SPECIES"}},
+)
+def iris_dataset_partitioned(context) -> pd.DataFrame:
+    partition = partition = context.partition_key.keys_by_dimension
+    species = partition["species"]
+    date = partition["date"]
+
+    # get_iris_data_for_date fetches all of the iris data for a given date,
+    # the returned dataframe contains a column named 'time' with that stores
+    # the time of the row as an integer of seconds since epoch
+    full_df = get_iris_data_for_date(date)
+
+    return full_df[full_df["species"] == species]
+
+
+@asset
+def iris_cleaned(iris_dataset_partitioned: pd.DataFrame):
+    return iris_dataset_partitioned.dropna().drop_duplicates()
+```
+
+Dagster uses the `partition_expr` metadata to craft the `SELECT` statement when loading the correct partition in a downstream asset. For multi-partitions, Dagster concatenates the `WHERE` statements described in the above sections to craft the correct `SELECT` statement.
+
+A partition must be selected when materializing the above assets, as described in [Materializing partitioned assets](/concepts/partitions-schedules-sensors/partitions#materializing-partitioned-assets). For example, when materializing the `2023-01-02|Iris-setosa` partition of the above assets, the following query will be used:
+
+```sql
+SELECT *
+ WHERE SPECIES in ('Iris-setosa')
+   AND TO_TIMESTAMP(TIME) >= '2023-01-02 00:00:00'
+   AND TO_TIMESTAMP(TIME) < '2023-01-03 00:00:00'
+```
+
+</TabItem>
+</TabGroup>
+
+---
+
+## Storing tables in multiple schemas
+
+You may want to have different assets stored in different DuckDB schemas. The DuckDB I/O manager allows you to specify the schema in several ways.
+
+If you want all of your assets to be stored in the same schema, you can specify the schema as configuration to the I/O manager, as we did in [Step 1: Configure the DuckDB I/O manager](/integrations/duckdb/using-duckdb-with-dagster#step-1-configure-the-duckdb-io-manager) of the [Using Dagster with DuckDB tutorial](/integrations/duckdb/using-duckdb-with-dagster).
+
+If you want to store assets in different schemas, you can specify the schema as part of the asset's asset key:
```

Comment:
yeah so it's `key` for SourceAssets since the `key` needs to be the full asset key. For normal `assets` it's `key_prefix` because the full asset key is `key_prefix`+`asset_name`. does that make sense? not sure i explained that well

Reply:
i added some additional explanation to say which value in the asset key becomes the prefix. let me know if that makes sense